src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs

```diff
@@ -37,9 +49,40 @@ internal sealed class OtlpProtobufSerializer
 
     internal IMetricDataTransport MetricDataTransport;
 
-    public OtlpProtobufSerializer(IMetricDataTransport metricDataTransport)
+    public OtlpProtobufSerializer(IMetricDataTransport metricDataTransport, ConnectionStringBuilder connectionStringBuilder, IReadOnlyDictionary<string, object> prepopulatedMetricDimensions)
     {
         this.MetricDataTransport = metricDataTransport;
+
+        // Taking a arbitrary number here for writing attributes.
+        byte[] temp = new byte[20000];
+        if (prepopulatedMetricDimensions != null)
+        {
+            // Initialize numberDataPoint attributes.
+            int cursor = 0;
+            SerializeTags(temp, ref cursor, prepopulatedMetricDimensions, FieldNumberConstants.NumberDataPoint_attributes);
+            this.prepopulatedNumberDataPointAttributes = new byte[cursor];
+            Array.Copy(temp, this.prepopulatedNumberDataPointAttributes, cursor);
+            this.prepopulatedNumberDataPointAttributesLength = cursor;
+
+            // Initialize histogramDataPoint attributes.
+            cursor = 0;
+            SerializeTags(temp, ref cursor, prepopulatedMetricDimensions, FieldNumberConstants.HistogramDataPoint_attributes);
+            this.prepopulatedHistogramDataPointAttributes = new byte[cursor];
+            Array.Copy(temp, this.prepopulatedHistogramDataPointAttributes, cursor);
+            this.prepopulatedHistogramDataPointAttributesLength = cursor;
+
+            // TODO: exponential histogram.
+        }
+
+        if (connectionStringBuilder != null && connectionStringBuilder.Namespace != null)
```

Comment:
Add a try/catch block as accessing `Namespace` and `Account` from `ConnectionStringBuilder` could throw an exception when connectionString does not contain them.

Reply:
added